src/sidebar/components/test/search-input-test.js

```diff
@@ -1,68 +1,88 @@
+    const FakeSpinner = () => null;
+    FakeSpinner.displayName = 'Spinner';
+
+    SearchInput.$imports.$mock({
+      './spinner': FakeSpinner,
+      '../store/use-store': callback => callback(fakeStore),
     });
   });
 
-  it('displays the search query', function() {
-    const el = util.createDirective(document, 'searchInput', {
-      query: 'foo',
-    });
-    const input = el.find('input')[0];
-    assert.equal(input.value, 'foo');
+  afterEach(() => {
+    SearchInput.$imports.$restore();
   });
 
-  it('invokes #onSearch() when the query changes', function() {
-    const onSearch = sinon.stub();
-    const el = util.createDirective(document, 'searchInput', {
-      query: 'foo',
-      onSearch: {
-        args: ['$query'],
-        callback: onSearch,
-      },
-    });
-    const input = el.find('input')[0];
-    const form = el.find('form');
-    input.value = 'new-query';
-    form.submit();
-    assert.calledWith(onSearch, 'new-query');
+  it('displays the active query', () => {
+    const wrapper = createSearchInput({ query: 'foo' });
+    assert.equal(wrapper.find('input').prop('value'), 'foo');
   });
 
-  describe('loading indicator', function() {
-    it('is hidden when there are no API requests in flight', function() {
-      const el = util.createDirective(document, 'search-input', {});
-      const spinner = el[0].querySelector('spinner');
+  it('resets input field value to active query when active query changes', () => {
+    const wrapper = createSearchInput({ query: 'foo' });
 
-      fakeStore.isLoading.returns(false);
-      el.scope.$digest();
+    // Simulate user editing the pending query, but not committing it.
+    typeQuery(wrapper, 'pending-query');
 
-      assert.equal(util.isHidden(spinner), true);
-    });
+    // Check that the pending query is displayed.
+    assert.equal(wrapper.find('input').prop('value'), 'pending-query');
 
-    it('is visible when there are API requests in flight', function() {
-      const el = util.createDirective(document, 'search-input', {});
-      const spinner = el[0].querySelector('spinner');
+    // Simulate active query being reset.
+    wrapper.setProps({ query: '' });
 
-      fakeStore.isLoading.returns(true);
-      el.scope.$digest();
+    assert.equal(wrapper.find('input').prop('value'), '');
+  });
 
-      assert.equal(util.isHidden(spinner), false);
-    });
+  it('invokes `onSearch` with pending query when form is submitted', () => {
+    const onSearch = sinon.stub();
+    const wrapper = createSearchInput({ query: 'foo', onSearch });
+    typeQuery(wrapper, 'new-query');
+    wrapper.find('form').simulate('submit');
+    assert.calledWith(onSearch, { $query: 'new-query' });
+  });
+
+  it('shows loading indicator when app is in a "loading" state', () => {
+    fakeStore.isLoading.returns(true);
+    const wrapper = createSearchInput();
+    assert.isTrue(wrapper.exists('Spinner'));
+  });
+
+  it('hides search button when app is in "loading" state', () => {
```

Comment:
Would it be more accurate to indicate that the button isn't even _rendered?_ (When I read "hide", I think CSS hidden). Ditto for the next test.

Reply:
"Does not render" would be more precise, yes. I was being a bit terse. On the other hand, "shows" and "hides" is accurate from the user's point of view 🤔